uwblueprint/extend-a-family
# Prototype for Create Module Page

## Notion ticket link

[Create Prototype](https://www.notion.so/uwblueprintexecs/Prototype-Page-Layout-Authoring-abdf45f2e1cf45edbd5ced7edabf6b7c?pvs=4)



## Implementation description
* Implemented a drag and drop grid with resizable / editable custom components that will serve as the base template for our create page.
* Implemented a reusable confirmation modal



## Steps to test
1. Head to "Prototype Page"
2. Drag Text Box / Match Box Source box (blue outline + "drag me") into the grid
3. A 1 x 1 Box should be placed into the grid
4. Resize and move the box around, box should line up perfectly with the grid in the background
5. Drag a "Text Box" into the grid, then click on it once to enter edit mode, try adjusting font weight and text alignment
6. Select an item again to deselect
7. Drag any item into the grid and select it, press delete, a confirmation modal should pop up warning you. If you accept, the item should delete, otherwise it should remain on the grid. 
8. Click on the eye icon to enter view mode, you should not see border or be able to move the elements, click again to re-enter edit mode and continue to make changes.




## What should reviewers focus on?
* How does the responsiveness feel?
* What feels uncomfortable or can be improved ?


## Checklist
